src/aws/nodemanager/nodemanager.py

```diff
@@ -54,19 +55,23 @@ async def run_task_pool(self):
         try:
             while True:
                 self.generate_heartbeat()
-                number_of_workers=len(self.available_workers)
-                remaining_tasks=len(self._tasks)%number_of_workers
-                task_per_worker=round((len(self._tasks)-remaining_tasks)/number_of_workers)
-                first_assignment=0
-                remainders_added=0
+                number_of_workers = len(self.available_workers)
```

Comment:
`self.available_workers` does not exist anymore. However, from looking at your functionality, I would expect to replace it with `available_workers = list(self.workers_running.keys()) + list(self.workers_pending.keys())` and then using `available_workers` instead of the self version would fix it.